trurl.c

```diff
@@ -184,6 +187,32 @@ static void show_version(void)
   curl_version_info_data *data = curl_version_info(CURLVERSION_NOW);
   fprintf(stdout, "%s version %s libcurl/%s [built-with %s]\n",
           PROGNAME, TRURL_VERSION_TXT, data->version, LIBCURL_VERSION);
+  /* puny code isn't guaranteed based on the version, so it must be polled
+   * from libcurl */
+  bool supports_puny = false;
+#ifdef SUPPORTS_PUNYCODE
+  const char *const *feature_name = data->feature_names;
+  while(*feature_name && !supports_puny) {
+    supports_puny = !strncmp(*feature_name, "IDN", 3);
```

Comment:
```suggestion
    supports_puny = !strcmp("IDN", *feature_name);
```
This seems wrong, it will make `trurl` think libcurl has the "IDN" feature even if `*feature_name` is actually "IDNOSAUR" or some other string that starts with "IDN".
Why are you using this?

Reply:
Emanuele, I strive to have your ability to see edge cases in code. I had no reason to do it one way or another, I was just thinking "IDN is only 3 characters so why would I check more?" Thank you :)

Reply:
Thank you :)

> "IDN is only 3 characters so why would I check more?"

Note that `strncmp("IDN", *feature_name, 4)` would have been equivalent to `strcmp("IDN", *feature_name, 3)`, but 3 isn't, it is effectively checking if `*feature_name` starts with "IDN".
Also note that there is no benefit in using `strncmp("foo", str, 4)` over `strcmp("foo", str)`, `strcmp` stops at the first NUL (`'\0'`) character, and `"IDN"[4] == '\0'`.

Reply:
Ahh, that makes sense. my embedded systems professor told us to use strncmp whenever possible because it may go off the end of arrays (if there is no null terminator), so it's mostly force of habit and I haven't done lots of string manipulation/parsing stuff like what we're doing. but hey, now I know lol